tower-balance/src/lib.rs

```diff
@@ -26,6 +26,11 @@ pub mod load;
 pub use choose::Choose;
 pub use load::Load;
 
+#[macro_use]
+mod macros {
+    include! { concat!(env!("CARGO_MANIFEST_DIR"), "/../gen_errors.rs") }
```

Comment:
In case anyone is curious about why the macro for generating error types is defined in a bare Rust file in the crate root and then `include!`ed into every crate that uses it, here's my rationale:

I wanted to be able to reuse the macro across multiple crates without having to copy it into all of them. However, I did _not_ want it to be part of a public API surface that we'd have to maintain. 

If the macro was in its own crate that the other crates depended on, and we didn't want to publish the macro to crates.io (making it public), it would have to be depended on as a path dependency. Crates with path dependencies cannot be published to crates.io, which would be a problem when we eventually release Tower to crates.io. Additionally, people would still be able to pull in the macro as a Git dependency. 

While it would be theoretically possible to add the bare .rs file as a dependency (such as by including the Tower repo as a submodule and `include!`ing the macro file), there's enough friction that I feel like it's not really as big a worry.  Thus, while this approach is admittedly somewhat ugly, it's the only way I could think of to reuse the macro without inadvertently making it public API.

As to why the macro is `include!`ed into its own module: there are two helper macros (`ignore_t!` and `arm!`) that are used inside the main macro for generating error types. I didn't want to expose those inside the crates that consume the macro, as they don't really make any sense outside of the context of the main macro. By including the macro in a module with `#[macro_use]` and only marking the main macro as `#[macro_use]`, crates using the macro can't access the helper macros.